### PR TITLE
Improvements and corrections for check constraints

### PR DESCRIPTION
**Pull request by zachmu**
Biggest changes:

1) Better validation
2) Checks in CREATE TABLE statements work as expected now (poor validation before)
3) ALTER TABLE DROP CONSTRAINT works as expected (doesn't assume a foreign key)
4) Unnamed checks are expected to be named by integrators now

### CODE REVIEW

**Comment by zachmu** [comment]
Constraints weren't actually being enforced, just warned. This matched previous versions of MySQL, not 8.x. So I changed it to actually kill the operation.

Also added the same functionality for update statements.

**Comment by zachmu** [comment]
This is still missing SHOW CREATE TABLE support, easy to add next

File: sql/analyzer/check_constraints.go [hunk 1]
```diff
@@ -134,39 +178,38 @@ func loadChecksFromTable(ctx *sql.Context, table sql.Table) ([]sql.Expression, e
...
-			}
+			loadedChecks = append(loadedChecks, constraint)
 		}
 	}
 	return loadedChecks, nil
 }
 
 func convertCheckDefToConstraint(ctx *sql.Context, check *sql.CheckDefinition) (*sql.CheckConstraint, error) {
-	parsed, err := sqlparser.ParseStrictDDL(check.AlterStatement)
+	parseStr := fmt.Sprintf("select %s", check.CheckExpression)
```
Comment (max-hoffman) [inline sql/analyzer/check_constraints.go:188]: much nicer

File: sql/plan/alter_check.go [hunk 2]
```diff
@@ -197,30 +188,35 @@ func (p *DropCheck) WithChildren(children ...sql.Node) (sql.Node, error) {
...
 	_ = pr.WriteChildren(fmt.Sprintf("Table(%s)", p.UnaryNode.Child.String()))
 	return pr.String()
 }
 
 func NewCheckDefinition(check *sql.CheckConstraint) (*sql.CheckDefinition, error) {
-	enforced := ""
-	if !check.Enforced {
-		enforced = " NOT ENFORCED"
+	// When transforming an analyzed CheckConstraint into a CheckDefinition (for storage), we strip off any table
+	// qualifiers that got resolved during analysis. This is to naively match the MySQL behavior, which doesn't print
```
Comment (max-hoffman) [inline sql/plan/alter_check.go:204]: cool, I was wondering how to remove those

File: sql/plan/alter_table.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+func (d *DropConstraint) String() string {
+	tp := sql.NewTreePrinter()
+	_ = tp.WriteNode("DropConstraint(%s)", d.Name)
+	_ = tp.WriteChildren(d.UnaryNode.Child.String())
+	return tp.String()
+}
+
+func (d *DropConstraint) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	panic("DropConstraint is a placehold node, but RowIter was called")
```
Comment (max-hoffman) [inline sql/plan/alter_table.go:34]: placehold->placeholder?

**Pull request merged**